Update key value processor documentation with new config options (#7413)
opensearch-project/documentation-website: _data-prepper/pipelines/configuration/processors/key-value.md

| tags_on_failure | When a `kv` operation causes a runtime exception within the processor, the operation is safely stopped without crashing the processor, and the event is tagged with the provided tags. | If `tags_on_failure` is set to `["keyvalueprocessor_failure"]`, `{"tags": ["keyvalueprocessor_failure"]}` will be added to the event's metadata in the event of a runtime exception. |
| value_grouping | Specifies whether to group values using predefined value grouping delimiters: `{...}`, `[...]', `<...>`, `(...)`, `"..."`, `'...'`, `http://... (space)`, and `https:// (space)`. If this flag is enabled, then the content between the delimiters is considered to be one entity and is not parsed for key-value pairs. Default is `false`. If `value_grouping` is `true`, then `{"key1=[a=b,c=d]&key2=value2"}` parses to `{"key1": "[a=b,c=d]", "key2": "value2"}`. |
| drop_keys_with_no_value | Specifies whether keys should be dropped if they have a null value. Default is `false`. If `drop_keys_with_no_value` is set to `true`, then `{"key1=value1&key2"}` parses to `{"key1": "value1"}`. |

> | strict_grouping | Specifies whether strict grouping should be enabled when `value_grouping` or `string_literal_character` options are used. Default is `false`. | If enabled, groups with unmatched end characters results in error. The event is ignored after error is logged.|

Reviewer: ```suggestion
| strict_grouping | Specifies whether strict grouping should be enabled when `value_grouping` or `string_literal_character` options are used. Default is `false`. | If enabled, groups with unmatched end characters results in errors. The event is ignored after error is logged. |
```

> | strict_grouping | Specifies whether strict grouping should be enabled when `value_grouping` or `string_literal_character` options are used. Default is `false`. | If enabled, groups with unmatched end characters results in errors. The event is ignored after error is logged. |

Reviewer: ```suggestion
| strict_grouping | Specifies whether strict grouping should be enabled when `value_grouping` or `string_literal_character` options are used. Default is `false`. | If enabled, groups with unmatched end characters results in errors. The event is ignored after errors are logged. |
```

> | strict_grouping | Specifies whether strict grouping should be enabled when `value_grouping` or `string_literal_character` options are used. Default is `false`. | If enabled, groups with unmatched end characters results in errors. The event is ignored after errors are logged. |

Reviewer 2: ```suggestion
| strict_grouping | Specifies whether strict grouping should be enabled when the `value_grouping` or `string_literal_character` options are used. Default is `false`. | When enabled, groups with unmatched end characters yield errors. The event is ignored after the errors are logged. |
```

> | string_literal_character | A single quote(`'`) or double quote(`"`) can be configured as string literal character. Default is Null. | If this option is used, the text between the single/double quote is ignored and is not used for key value parsing. An input like `text1 "key1=value1" text2 key2=value2` parses to `{"key2": "value2"}`.| 

Reviewer: ```suggestion
| string_literal_character | Can be set to either a single quotation mark (`'`) or a double quotation mark (`"`). Default is `null`. | When this option is used, any text contained within the specified quotation mark character will be ignored and excluded from key-value parsing. For example, `text1 "key1=value1" text2 key2=value2` would parse to `{"key2": "value2"}`. | 
```

> | key_value_when | A [conditional expression](https://opensearch.org/docs/latest/data-prepper/pipelines/expression-syntax/), such as `/some-key == "test"'`. that will be evaluated to determine whether the processor will be run on the event. |

Reviewer: ```suggestion
| key_value_when | Allows you to specify a [conditional expression](https://opensearch.org/docs/latest/data-prepper/pipelines/expression-syntax/), such as `/some-key == "test"`, which will be evaluated to determine if the processor should be applied to the event or not. |
```

> | key_value_when | Allows you to specify a [conditional expression](https://opensearch.org/docs/latest/data-prepper/pipelines/expression-syntax/), such as `/some-key == "test"`, which will be evaluated to determine if the processor should be applied to the event. |

Reviewer 2: ```suggestion
| key_value_when | Allows you to specify a [conditional expression](https://opensearch.org/docs/latest/data-prepper/pipelines/expression-syntax/), such as `/some-key == "test"`, that will be evaluated to determine whether the processor should be applied to the event. |
```
